### PR TITLE
Properly the count the number of available CPU for autogen_mapping script

### PR DESCRIPTION
**Pull request by felipeZ**
## Changes
* solve #688

### CODE REVIEW

File: scripts/xtp_autogen_mapping.in [hunk 1]
```diff
@@ -75,10 +75,9 @@ def search_cores() -> int:
         return multiprocessing.cpu_count()
 
     with open(cpuinfo, 'r') as handler:
-        for line in handler:
-            if "cpu cores" in line:
-                break
-    return int(line.split()[-1])
+        lines = handler.readlines()
+
+    return len(list(filter(lambda x: "processor" in x, lines)))
```
Comment (JensWehner) [inline scripts/xtp_autogen_mapping.in:80]: it does what it says, it counts how often processor appears in the file. ?

**Review by JensWehner — APPROVED** [review]
Beautiful code does not look like python at all. Thanks

File: scripts/xtp_autogen_mapping.in (hunk 1 above)
Comment (felipeZ) [reply scripts/xtp_autogen_mapping.in:80]: yes, exactly.
  have to transform the filter which is an `iterator` into a `list` to be able to count the elements

**Comment by baumeier** [comment]
Does this now manage to detect if hyperthreading is on or not? If one counts those as well, it causes havoc with orca.

File: scripts/xtp_autogen_mapping.in (hunk 1 above)
Comment (JensWehner) [reply scripts/xtp_autogen_mapping.in:80]: really beautiful.. :)

**Comment by felipeZ** [comment]
@baumeier  I have added the functionality to check for hyperthreading

**Review by JensWehner — APPROVED** [review]

**Review by baumeier — APPROVED** [review]

**Pull request merged**